controlpanel/frontend/forms.py

```diff
@@ -125,3 +125,15 @@ def __init__(self, *args, **kwargs):
         else:
             self.fields['datasource'].queryset = S3Bucket.objects.all()
 
+
+class CreateParameterForm(forms.Form):
+    key = forms.CharField(
+        validators=[RegexValidator(r'[a-zA-Z0-9_]{1,50}')]
+    )
+    role_name = forms.CharField(
+        validators=[RegexValidator(r'[a-zA-Z0-9_-]{1,60}')]
+    )
+    value = forms.CharField(widget=forms.PasswordInput)
+    app_type = forms.ChoiceField(
+        choices=Parameter._meta.get_field('app_type').choices
```

Comment:
Is this the recommended way to get the field choices? Accessing a "private" property seems wrong.

Reply:
I have set as a constant in the models file. That is fairly standard